src/main/java/fi/okm/jod/yksilo/service/koski/KoskiService.java

```diff
@@ -159,4 +172,25 @@ private static LocalDate getLocalDate(JsonNode nimet, String path) {
     JsonNode node = readJsonProperty(nimet, path);
     return node != null && node.isTextual() ? LocalDate.parse(node.asText()) : null;
   }
+
+  @Cacheable(value = "osaamisetIdentified", key = "'user_' + #user.id")
+  @Transactional(readOnly = true)
+  public List<KoulutusDto> getOsaamisetIdentified(JodUser user, List<UUID> uuids) {
+    if (uuids.size() > (Limits.KOULUTUSKOKONAISUUS * Limits.KOULUTUS_PER_KOKONAISUUS)) {
+      throw new KoskiServiceException("UUID limit exceeded.");
+    }
+
+    var koulutusList = koulutusRepository.findAllById(uuids);
+    if (koulutusList.isEmpty()) {
+      return Collections.emptyList();
+    }
+    var userId = user.getId();
+    for (var koulutus : koulutusList) {
+      if (!userId.equals(koulutus.getYksilo().getId())) {
+        throw new KoskiServiceException(
+            "Koulutus UUID " + koulutus.getId() + " belongs to another user.");
+      }
+    }
```

Comment:
I suggest creating a repository query method that includes the user id check; this is a common pattern in the service methods that access profile data, the query should filter out data that does not belong to the user.